src/cwts/networkanalysis/GradientDescentVOSLayoutAlgorithm.java

```diff
@@ -0,0 +1,397 @@
+     * @param requiredNQualityValueImprovements Required number of quality value
+     *            improvements
+     * @param random                            Random number generator
+     */
+    public GradientDescentVOSLayoutAlgorithm(int attraction, int repulsion, double edgeWeightIncrement,
+            int maxNIterations, double initialStepSize, double minStepSize, double stepSizeReduction,
+            int requiredNQualityValueImprovements, Random random)
+    {
+        super(attraction, repulsion, edgeWeightIncrement);
+
+        this.maxNIterations = maxNIterations;
+        this.initialStepSize = initialStepSize;
+        this.minStepSize = minStepSize;
+        this.stepSizeReduction = stepSizeReduction;
+        this.requiredNQualityValueImprovements = requiredNQualityValueImprovements;
+        this.random = random;
+    }
+
+    /**
+     * Clones the algorithm.
+     *
+     * @return Cloned algorithm
+     */
+    public GradientDescentVOSLayoutAlgorithm clone()
+    {
+        GradientDescentVOSLayoutAlgorithm gradientDescentVOSLayoutAlgorithm;
+
+        gradientDescentVOSLayoutAlgorithm = (GradientDescentVOSLayoutAlgorithm)super.clone();
+
+        return gradientDescentVOSLayoutAlgorithm;
+    }
+
+    /**
+     * Returns the maximum number of iterations.
+     *
+     * @return Maximum number of iterations
+     */
+    public int getMaxNIterations()
+    {
+        return maxNIterations;
+    }
+
+    /**
+     * Returns the initial step size.
+     *
+     * @return Initial step size
+     */
+    public double getInitialStepSize()
+    {
+        return initialStepSize;
+    }
+
+    /**
+     * Returns the minimum step size.
+     *
+     * @return Minimum step size
+     */
+    public double getMinStepSize()
+    {
+        return minStepSize;
+    }
+
+    /**
+     * Returns the step size reduction.
+     *
+     * @return Step size reduction
+     */
+    public double getStepSizeReduction()
+    {
+        return stepSizeReduction;
+    }
+
+    /**
+     * Returns the required number of quality value improvements.
+     *
+     * @return Required number of quality value improvements
+     */
+    public int getRequiredNQualityValueImprovements()
+    {
+        return requiredNQualityValueImprovements;
+    }
+
+    /**
+     * Sets the maximum number of iterations.
+     * 
+     * @param maxNIterations Maximum number of iterations
+     */
+    public void setMaxNIterations(int maxNIterations)
+    {
+        this.maxNIterations = maxNIterations;
+    }
+
+    /**
+     * Sets the initial step size.
+     * 
+     * @param initialStepSize Initial step size
+     */
+    public void setInitialStepSize(double initialStepSize)
+    {
+        this.initialStepSize = initialStepSize;
+    }
+
+    /**
+     * Sets the minimum step size.
+     * 
+     * @param minStepSize Minimum step size
+     */
+    public void setMinStepSize(double minStepSize)
+    {
+        this.minStepSize = minStepSize;
```

Comment:
Shouldn't the min step size be positive, and smaller than the initial step size?